arviz/data/inference_data.py

```diff
@@ -133,6 +133,18 @@ def __repr__(self):
             msg += "\n\nWarmup iterations saved ({}*).".format(WARMUP_TAG)
         return msg
 
+    def _repr_html_(self):
+        """Make html representation of object"""
+        xr.set_options(display_style="html")
```

Comment:
I am not sure about this. html repr is the default for xarray now (we could require a version where this is already default) so this will generally have no effect, and if it were not the default, it means the user has specifically changed it so undoing the change in the background and with no option to override it seems like a bad idea.

How does xarray choose between html repr and "regular" print? Maybe we can check xarray defaults us too to choose one or the other?

Reply:
I understand the issue, I tried to solve it by saving the default style and reapplying after the display of html_repr.
xarray is doing like [this](https://github.com/jsignell/xarray/blob/1d960933ab252e0f79f7e050e6c9261d55568057/xarray/core/common.py#L138) The default option is "text".

Reply:
I would mimic xarray behaviour. If display style is text we should not print the html repr. 

The default in latest xarray is already html https://github.com/pydata/xarray/blob/master/xarray/core/options.py#L23 so we should probably require xarray>=0.15.1. I think we should have html repr by default so requiring this or later xarray version would make it default.

Reply:
Okay, I understand. I was reading the fork of the guy who implemented the html_repr for xarray a long time ago.

Reply:
could it be you modified that and rolled back the changes? I have a vague memory of `f"<pre>{escape(repr(self))}</pre>"` but now I an not sure. Either way, this clearly has to be changed to use the xarray default instead of overriding it

Reply:
Yes, that's the case. I apologize for these discrepancies. I have corrected the ones I found.